### ISSUE TITLE
Support for `render` prop on the composed component

### ISSUE THREAD

**Issue by ctrlplusb** [pedronauck/react-adopt#17]
Would you be open to the supporting a `render` prop as well as the current `children` prop for providing the rendering function?

```jsx
const Composed = adopt(...);

<Composed render={({ greet, name }) => <div>{greet} {name}</div>} />
```

**Comment by pedronauck** [comment]
Sure, I think that can be a good thing to do...

**Comment by pedronauck** [comment]
Sure @ctrlplusb, great idea 🙏 
I released this feature on [v0.5.1](https://github.com/pedronauck/react-adopt/releases/tag/v0.5.1) 🚀

**Comment by ctrlplusb** [comment]
Awesome 🎉
